File: ki/paiza3.py
N, K = map(int,input().split())
adjacent_list = [ [] for i in range(N)]
# print(adjacent_list)
for i in range(N - 1):
  a, b = map(int, input().split())
  adjacent_list[a - 1].append(b)
  adjacent_list[b - 1].append(a)

for i in range(K):
  c, d = map(int, input().split())
  if d in adjacent_list[c - 1]:
    print('YES')
  else:
    print('NO')

File: sennkeitansaku2/paiza3.swift
import Foundation 

let input = readLine()!.split(separator: " ").map { Int($0)! }
let (n, m) = (input[0], input[1])

let a = readLine()!.split(separator: " ").map { Int($0)! } 
let b = readLine()!.split(separator: " ").map { Int($0)! }

var indexA = 0 
var indexB = 0 

 while indexA < n && indexB < m {
        if a[indexA] == b[indexB] {
            indexB += 1
        }
        indexA += 1
    }

    let result = indexB == m ? "Yes" : "No"
    
print(result)